src/main/java/org/openstreetmap/atlas/geography/boundary/CountryToShardListCache.java

```diff
@@ -0,0 +1,65 @@
+package org.openstreetmap.atlas.geography.boundary;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.geography.sharding.SlippyTile;
+import org.openstreetmap.atlas.streaming.resource.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Country to shard list that is read from a .txt file. Useful for when using dense shard trees to
+ * reduce overhead of recalculating country/shard intersection.
+ *
+ * @author james-gage
+ */
+public final class CountryToShardListCache
+{
+    private static final Logger logger = LoggerFactory.getLogger(CountryToShardListCache.class);
+    private final HashMap<String, List<SlippyTile>> countryToShards = new HashMap<>();
+
+    public static CountryToShardListCache create(final Resource file)
```

Comment:
Throwing a random idea out here - would it be useful to have the CountryToShardListCache also take in a CountryBoundaryMap and Sharding?  Then CountryToShardListCache can support read() and write() methods and can essentially become a resource.  Not sure what the scope of this very useful functionality is though!  

Another advantage is that it would also make the CountryToShardListCacher really only a thin layer.  The logic in the CountryToShardListCacher can then be in one place and users won't have to duplicate that code to write CountryToShardList files.

Reply:
Great idea, working on this now